packages/venia-concept/src/components/ProductOptions/tile.js

```diff
@@ -27,15 +33,20 @@ class Tile extends Component {
             hasFocus,
             isSelected,
             item,
-            // eslint-disable-next-line
-            itemIndex,
-            ...restProps
+            onBlur,
+            onClick,
+            onFocus
         } = this.props;
         const className = classes[getClassName('root', isSelected, hasFocus)];
         const { label } = item;
 
         return (
-            <button {...restProps} className={className}>
+            <button
+                className={className}
+                onBlur={onBlur}
+                onClick={onClick}
+                onFocus={onFocus}
```

Comment:
This is curious. What's the goal here?

Reply:
ListItems contain a prop called `itemIndex` which is relevant for some things that can be items and  not relevant for others such as `tile` or `swatch`. I switched from spreading the rest to only using the specific props here because it felt more intentional. The alternative would be to delete the `itemIndex` from the `restProps` before spreading, but that felt weird too.

Reply:
The problem with picking props to intentionally apply is you're creating a very arbitrary whitelist. Deleting a single prop might feel clumsy, but it's not leaky.

The pattern originally in place excluded `itemIndex` from `restProps` without having to use the `delete` operator, without excluding all kinds of other props: data attributes, `id`, etc.

Reply:
Sidebar — just following this via emails, it was a little confusing reading `restProps` and thinking REST as in API. I'm not sure if that's a conventional name, but was thinking I wouldn't have been confused by `otherProps` after seeing the context.

Reply:
@brendanfalkowski the idea is that those are the "rest of the props" and yeah, it's sort of a convention by virtue of the feature's name. See [rest parameters](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters).